lib/Runtime/Base/ThreadContext.cpp

```diff
@@ -1132,7 +1136,7 @@ ThreadContext::AddPropertyRecordInternal(const Js::PropertyRecord * propertyReco
 #if DBG
     // Only Assert we can find the property if we are not adding a symbol.
     // For a symbol, the propertyName is not used and we won't be able to look the pid up via name.
-    if (!propertyRecord->IsSymbol())
+    if (propertyNameLength && !propertyRecord->IsSymbol())
```

Comment:
Same thing here- the invariant should be that if we add a property record, it should be in the map when we try to retrieve it (if you added the check for perf, its probably not worth it since this is debug only code)